Branch-SDK-Automation-TestBed/src/main/java/io/branch/branchandroiddemo/activities/MainActivity.java

```diff
@@ -0,0 +1,119 @@
+import io.branch.branchandroiddemo.Constants;
+import io.branch.branchandroiddemo.R;
+import io.branch.branchandroiddemo.TestData;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+    private Button  btCreateDeepLink, btNativeShare, btTrackUser,
+             btCreateQrCode, btSetDMAParams, btLogEvent, btReadLogs, btInitSession;
+    private TextView tvMessage, tvUrl;
+    ToggleButton trackingCntrlBtn;
+
+    BranchWrapper branchWrapper;
+    protected void initialize(){
+        btCreateDeepLink = findViewById(R.id.bt_create_deep_link);
+        btNativeShare = findViewById(R.id.bt_native_share);
+        tvMessage = findViewById(R.id.tv_message);
+        tvUrl = findViewById(R.id.tv_url);
+        btCreateQrCode = findViewById(R.id.bt_create_qr_code);
+        btSetDMAParams = findViewById(R.id.bt_set_dma_params);
+        btLogEvent = findViewById(R.id.bt_logEvent_with_callback);
+        btReadLogs = findViewById(R.id.bt_Read_Logs);
+        btTrackUser = findViewById(R.id.bt_track_user);
+        btInitSession = findViewById(R.id.bt_init_session);
+        trackingCntrlBtn = findViewById(R.id.tracking_cntrl_btn);
+
+        btCreateDeepLink.setOnClickListener(this);
+        btNativeShare.setOnClickListener(this);
+        btCreateQrCode.setOnClickListener(this);
+        btSetDMAParams.setOnClickListener(this);
+        btLogEvent.setOnClickListener(this);
+        btReadLogs.setOnClickListener(this);
+        btTrackUser.setOnClickListener(this);
+        btInitSession.setOnClickListener(this);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        Common.getInstance().clearLog();
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        initialize();
+
+        branchWrapper = new BranchWrapper(getApplicationContext());
+        branchWrapper.delayInitializationIfRequired(getIntent());
+
+        Branch.enableLogging(BranchLogger.BranchLogLevel.VERBOSE);
+
+        trackingCntrlBtn.setChecked(Branch.getInstance().isTrackingDisabled());
+        trackingCntrlBtn.setOnCheckedChangeListener((buttonView, isChecked) -> Branch.getInstance().disableTracking(isChecked));
+    }
+
+    @Override
+    public void onClick(View view) {
+
+        /*  getIntent().putExtra("testData", "{\"BUOData\":{\"imageUrl\":\"https:\\/\\/www.qentelli.com\\/sites\\/default\\/files\\/qentelli-logo.svg\",\"canonicalIdentifier\":\"content\\/2\",\"contentTitle\":\"Sample Content Title\",\"contentDesc\":\"Sample Content Description\"},\"Include\":\"Yes\",\"CreateLinkReference\":{\"feature\":\"sharing\",\"stage\":\"new user\",\"desktopUrl\":\"www.qentelli.com\",\"androidUrl\":\"https:\\/\\/play.google.com\\/store\\/apps\\/details?id=com.instagram.android\",\"campaign\":\"content 2 launch\",\"channelName\":\"instagram\",\"additionalData\":\"When all fields provided with valid entries\",\"iOSUrl\":\"www.qentelli.com\"},\"TrackContentData\" : {\n" +
```

Comment:
Is commented code needed? Otherwise delete

Reply:
Done.